chromium_src/components/external_intents/android/intercept_navigation_delegate_impl.cc

```diff
@@ -23,13 +21,15 @@ namespace {
 using navigation_interception::InterceptNavigationDelegate;
 using navigation_interception::NavigationParams;
 
-Profile* GetOriginalProfile() {
-  return ProfileManager::GetActiveUserProfile()->GetOriginalProfile();
-}
-
 bool ShouldPlayVideoInBrowser(const GURL& url) {
-  if (!GetOriginalProfile()->GetPrefs()->GetBoolean(
-          kPlayYTVideoInBrowserEnabled)) {
+  PrefService* pref_service =
```

Comment:
it is definitely not safe to have a singleton for the profile. We always need to be thinking ahead to Tor, etc... when android will have more than one profile. When `BraveInterceptNavigationDelegate` is created you can get the BrowserContext from the WebContents and then use `user_prefs::UserPrefs::Get`

Reply:
Addressed in https://github.com/brave/brave-core/commit/efe8a130ba7dc8967c3364ff8ee11d9c227898cf